File: backend/src/main/java/com/aiexam/service/ExamService.java
package com.aiexam.service;

import com.aiexam.dto.CreateExamRequest;
import com.aiexam.dto.ExamResponse;
import com.aiexam.dto.UpdateExamRequest;
import com.aiexam.entity.Exam;

import java.util.List;

public interface ExamService {

    ExamResponse createExam(CreateExamRequest request);

    ExamResponse updateExam(String examId, UpdateExamRequest request);

    ExamResponse getExamById(String examId);

    List<ExamResponse> getAllExams();

    List<ExamResponse> getMyExams();

    void deleteExam(String examId);

    ExamResponse publishExam(String examId);

    boolean isExamAvailable(String examId);

    Exam validateExamForAttempt(String examId);

    List<ExamResponse> getAvailableExams();
}

File: frontend/src/services/examService.js
import api from "../api/axios";

const examService = {

    getAllExams() {
        return api.get("/exams");
    },

    getExamById(id) {
        return api.get(`/exams/${id}`);
    },

    createExam(data) {
        return api.post("/exams", data);
    },

    updateExam(id, data) {
        return api.put(`/exams/${id}`, data);
    },

    deleteExam(id) {
        return api.delete(`/exams/${id}`);
    },

    publishExam(id) {
        return api.patch(`/exams/${id}/publish`);
    }

};

export default examService;